### PR TITLE
Convert frontend to Vue 2.7 composition api

### PR DESCRIPTION
**Pull request by peace-maker**
Turn on typescript in .vue components and switch to newest [composition API](https://vuejs.org/api/composition-api-setup.html) to define components as well as [<script setup>](https://vuejs.org/api/sfc-script-setup.html) to clean up the code.

This paves the way for Vue 3.

Refs #30

### CODE REVIEW

File: converters/pkappa2lib/converterfile.hexpat
Comment (spq) [inline converters/pkappa2lib/converterfile.hexpat:1]: is this related?

**Review by spq — APPROVED** [review]
LGTM % hexpat file

File: converters/pkappa2lib/converterfile.hexpat
Comment (peace-maker) [reply converters/pkappa2lib/converterfile.hexpat:1]: whops, no, but it's a required change for the current converter cache file format. I can move it out of this branch but my commit discipline isn't great right now :D
